Haz refactor

ppapasani1 opened the pull request:
PR to refactor FL hazard code.

- Modularize the FL hazard component. Optimize logic by cell(Instead of Locid)
- Added RMS Version of QuadKey conversion.
- Added hazard smoke tests for APFL, EUFL, USFL models. (Runs on both Linux and Windows)
![image](https://user-images.githubusercontent.com/37124509/97039546-753d7c80-1521-11eb-8cb1-d301514e7b9a.png)

- Large smoke test baseline is stored at `\\ca1fs01\me_dev\modcert\E2E_Smoketests` or `/me-dev/modcert/E2E_Smoketests`
- The new code will be structure into Module, Model, and tests as shown below. 
```
├── Model
│   ├── APFL
│   ├── EUFL
│   ├── NZFL
│   └── USFL
├── Module
│   └── hazard
└── tests
    └── resources
        ├── apfl_hazard_test
        ├── eufl_hazard_test
        ├── eufl_hazard_test2
        └── usfl_hazard_test
```

Note- Still old code is not removed. Will remove once complete integration is done in the upcoming PRs.

ppapasani1 commented:
@ssura-rms @rishikeshkumar9 Please keep an eye on this PR. 70-80% of the existing FL hazard code is refactored. Adding you both as reviewers.

zsun-rms commented on climateE2E/E2E_Prototype_Main.R:
```diff
@@ -9,10 +9,10 @@ library('plyr')
 # library(yaml)
 starttime=Sys.time()
-isDockerRun <- FALSE
+isDockerRun <- T
```
Is there a place for windows runs to set it to false?

ppapasani1 replied:
Will remove this in the final clean. These variables are being used when running against the old code.

zsun-rms commented on climateE2E/HD_Climate_E2E.R:
```diff
@@ -0,0 +1,47 @@
+isDockerRun <- T
+if(interactive()) {
+  codeDir <- 'D:/Pruthvi/gitProjects/ClimateModel_E2E/climateE2E/climateE2E/'
+  configfile <- if(isDockerRun) "D:/Pruthvi/gitProjects/ClimateModel_E2E/climateE2E/climateE2E/tests/resources/usfl_hazard_test/config.json" else "config.yml"
```
E2E should support both `yml` and `json` format for loading config file, not that docker run accept `json` format else accept 'yml'

ppapasani1 replied:
Agree. It will be the JSON for all runtimes. I've already cleaned up in the subsequent PR.

sthupakula-rms commented on climateE2E/Model/EUFL/eufl_hazard.R:
```diff
@@ -0,0 +1,16 @@
+## Place holder to override Model specific setting
```
Such structure for models looks Good 👍

sthupakula-rms commented on climateE2E/tests/compare_dt.R:
```diff
@@ -0,0 +1,165 @@
+##### 25th of March 2020 ######################
+###############################################
+
+library(data.table)
```
May move to common library loader

sthupakula-rms commented on climateE2E/tests/fl_hazard_tests.R:
```diff
@@ -0,0 +1,65 @@
+}
+
+os <- get_os()
+ModelData <- if(os=="win") "//ca1fs01/mc_stage/NGDLMDATA" else "/NGDLMDATA"
```
Nice 👍

sthupakula-rms commented on climateE2E/tests/resources/apfl_hazard_test/config.json:
```diff
@@ -0,0 +1,29 @@
+{
+    "EDScsv": "D:/Pruthvi/gitProjects/ClimateModel_E2E/climateE2E/climateE2E/tests/resources/apfl_hazard_test/FlatExposure__batch0.csv",
```
Can we provide relative referencing here somehow, so that anyone can run the tests?

ppapasani1 replied:
Yes. I already fixed in the subsequent PR.

sthupakula-rms commented on climateE2E/tests/resources/apfl_hazard_test/config.json:
```diff
@@ -0,0 +1,29 @@
+        "userid": "sa",
+        "password": "Rmsuser!"
+        },
+    "OutFolder": "D:/Pruthvi/gitProjects/ClimateModel_E2E/climateE2E/climateE2E/tests/resources/apfl_hazard_test/TestResult",
```
same as above, and for rest of the tests.

ppapasani1 replied:
Yes. I already fixed it in the subsequent PR.

sthupakula-rms reviewed (changes requested):
We are in the right direction. More cleanup should be taken once the migration is structurally complete. A few remarks below.

ppapasani1 commented on climateE2E/Module/e2e_logger.R:
```diff
@@ -0,0 +1,31 @@
+##  - Close to log4j
+## Suggest to use log_debug statements in the code 
+
+if("logger" %in% (.packages())){
```
Yes, they will be loaded at the very initial level. This is just an additional check in case if we are calling logging from any other external codes, like for testing. 
For docker, we have a shell script that installs all the required packages, we need to add that package name to the shell script.

ppapasani1 commented on climateE2E/Module/hazard/fl_calculate.R:
```diff
@@ -0,0 +1,152 @@
+  return(rbindlist(Events))
+}
+
+combineFloodDepth <- function(events1d, events2d, eventsra, includeRA, useCorrectTimeStampMethodology){
```
Missing the return function. Will add it.

ppapasani1 commented:
> We are in the right direction. More cleanup should be taken once the migration is structurally complete. A few remarks below.

@sthupakula-rms I've addressed most of the comments in the follow-up PR. Can you please approve this?
